src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -13,18 +13,26 @@ internal class AwsClient( IAmazonSecurityTokenService stsClient ) : IAwsClient {
 		AwsRole role,
 		int durationInMinutes
 	) {
-		var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
-			PrincipalArn = role.PrincipalArn,
-			RoleArn = role.RoleArn,
-			SAMLAssertion = samlAssertion,
-			DurationSeconds = durationInMinutes * 60,
-		} );
 
-		return new AwsCredentials(
-			SessionToken: authResp.Credentials.SessionToken,
-			AccessKeyId: authResp.Credentials.AccessKeyId,
-			SecretAccessKey: authResp.Credentials.SecretAccessKey,
-			Expiration: authResp.Credentials.Expiration.ToUniversalTime()
-		);
+		try {
+			var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
+				PrincipalArn = role.PrincipalArn,
+				RoleArn = role.RoleArn,
+				SAMLAssertion = samlAssertion,
+				DurationSeconds = durationInMinutes * 60,
+			} );
+
+			return new AwsCredentials(
+				SessionToken: authResp.Credentials.SessionToken,
+				AccessKeyId: authResp.Credentials.AccessKeyId,
+				SecretAccessKey: authResp.Credentials.SecretAccessKey,
+				Expiration: authResp.Credentials.Expiration.ToUniversalTime()
+			);
+		} catch ( Exception ex ) {
+			if (ex.Message == "The requested DurationSeconds exceeds the MaxSessionDuration set for this role.") {
+				throw new BmxException( "Duration exceeds the MaxSessionDuration for this role. Lower it in config/parameter", ex );
```

Comment:
Do you know what the other errors are and are they just not useful or easy to understand?

Reply:
Yea I looked over the [rest of the errors](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithSAML.html#API_AssumeRoleWithSAML_Errors) and they seem pretty unlikely to happen and not too user friendly if I we're to add a case

Reply:
Some more general [common errors](https://docs.aws.amazon.com/STS/latest/APIReference/CommonErrors.html) here but most of them seem to be handled by how bmx is used. The only one that could be added I would think is ThrottlingException

Reply:
Sounds good 🤝

Reply:
Looking online I can't even seem to find anyone running into or an example of the ThrottlingException error for AssumeRole. I think it should be fine to leave out